### DISCUSSION TITLE
Issue process: Automation of labelling & moving issues

### DISCUSSION THREAD

**Discussion by Rumyra** [mdn/mdn-community#575] [Content]
As it stands new issues are processed initially by adding the `needs-triage` and content area labels (`Content:JS` `Content:CSS` etc...)

This then allows a defined team of issue triagers to come in and add more information about the issue, so eventually contributors can pick them up and 'fix' them. See https://developer.mozilla.org/en-US/docs/MDN/Contribute/Processes/Content_bug_triage for the full process.

The initial process is currently done by a human, when in fact it probably doesn't need to be. Labels can be automatically assigned when a new issue is raised. There's also cases we don't account for with the initial labels:
* Is it a platform issue (should be moved to yari)
* Is it a bcd issue (should be moved to browser compat)
* Other things like discussions and content requests etc... can also be moved (although this requires some other processes)

Things to discuss in regards to the initial labelling of issues:
1. How do we automatically add labels (I was originally considering yari, most prominently issues come from MDN pages via the 'Report a problem with this page on github' link and we then know which page is being referred to, and priority too) - happy to discuss gh actions for this.
2. How do we move issues, currently a peer (someone doing the triaging) needs write access to both the from repo and to repo
3. Do the labels we have make sense? After the triage process (second step above) a time is added, a priority, whether it's a good first task or if help is needed. This is so community contributors can have the information _they_ might need.

Sticking this here too for reference:
https://developer.mozilla.org/en-US/docs/MDN/Contribute/Fixing_MDN_content_bugs

**Comment by Ryuno-Ki** [comment]
Thanks for opening this discussion, @Rumyra.

I think, a good first step is to inquiry, whether the labels get used at all.
That is, you can filter GitHub issues for labels. Are the teams looking at these?
Are PRs within scope of this discussion?
Personally, I am rather going by the subject (and handle some typo fix PRs when I have slack time) right now.

**Reply by Rumyra** [reply]
Yes agreed @Ryuno-Ki it would be good to get feedback in this discussion about the current labels we have.

PRs is outside scope, but I will be bringing plenty more discussions around all these things :)

**Comment by dontcallmedom** [comment]
As mentioned elsewhere, I've been doing a kind of out-of-band labeling of issues to help identify those that are relevant to particular spec / group / org : https://dontcallmedom.github.io/mdn-issue-by-spec/ via https://github.com/dontcallmedom/mdn-issue-by-spec/

My motivation for this tool is that it's much easier for me (and I expect many others) to watch issues on which I know I can bring useful input than to watch the firehose of issues covering any Web platform related topic.

In the [Web Platform Tests project](https://github.com/web-platform-tests/wpt) (which has similarities with MDN in its Web-Platform wide scope and its potential needs of expertise for many different topics), this kind of information is maintained directly in the repo via labels.

If my use case of classifying issues along the line of specs or groups has a broad enough appeal, I would be happy to adapt my tool to make it set labels on the repo.

**Comment by pepelsbey** [comment]
Closing as answered/settled.
